tracetools_read/tracetools_read/trace.py

```diff
@@ -64,6 +67,38 @@ def get_trace_events(trace_directory: str) -> List[DictEvent]:
     return events
 
 
+def get_events_with_name(
+    event_name: str,
+    events: List[DictEvent],
+) -> List[DictEvent]:
+    """
+    Get all events with the given name.
+
+    :param event_name: the event name
+    :param events: the events to check
+    :return: the list of events with the given name
+    """
+    return [e for e in events if get_event_name(e) == event_name]
+
+
+def get_events_with_field_value(
+    field_name: str,
+    field_values: Any,
```

Comment:
It might be worthwhile to restrict this to `Iterable` types; then you could get rid of the `not isinstance(field_values, (list,set))` below.

But it is just a suggestion, feel free to ignore.

Reply:
The `Any` is meant to cover both a single value and lists/sets of values. I guess I could change callers to only provide a list/set and not an individual value, but this way the test code is a bit simpler (which was my intention).

Reply:
FYI I moved the two util functions here from `tracetools_read/trace.py` to `tracetools_read/__init__.py`: https://github.com/ros2/ros2_tracing/compare/cdab11e618ce64a089bdd391699da96894536d55..edeaa9454d15e05865b08d625f15e2c7ecbb8dc7. The reason is that the `babeltrace` module is not available on some systems (e.g., Windows), so it cannot be (in)directly imported by tests at the usual file/module level. It's imported indirectly in some test methods, since that is never in scope if LTTng is not installed (see `@unittest.skipIf`) and therefore never in scope if `babeltrace` is not available. Since these two functions don't need `babeltrace`, I moved them to `tracetools_read/__init__.py` with the rest of the util functions to simplify downstream imports.